Uninitialized variable usage in `resize__system_u...` in `@psystem.nim.c` in ORC

### Nim Version

```
Nim Compiler Version 2.2.4 [Linux: amd64]
Compiled at 2025-10-05
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: f7145dd26efeeeb6eeae6fff649db244d81b212d
active boot switches: -d:release
```

```
Nim Compiler Version 2.2.5 [Linux: amd64]
Compiled at 2025-10-05
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: cfe163795f3a8ac8ecf11dc0ab788a3758b2a32e
active boot switches: -d:release
```

```
Nim Compiler Version 2.3.1 [Linux: amd64]
Compiled at 2025-10-05
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: 483389d3999e243d7c61647c55187b0578dc57ee
active boot switches: -d:release
```

```
gcc (Debian 15.2.0-4) 15.2.0
Copyright (C) 2025 Free Software Foundation, Inc.
```

```

Debian clang version 19.1.7 (7)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-19/bin
```

### Description

An empty, 0-byte file as a Nim source file reproduces this, but not uniquely. So:
```nim
```
works. Insert some Nim code to taste; it doesn't really affect this.

Then the resulting `@psystem.nim.c` contains:
```c
static N_INLINE(NI, resize__system_u2271)(NI old_p0) {
	NI result;
{
		{
	if ((!((old_p0 <= ((NI) 0))))) {
	goto LA3_;
}
result = ((NI) 4);
}
goto LA1_;
LA3_: ;
{
	NI TM__Q5wkpxktOdTGvlSRo9bzt9aw_13;
if ((!((old_p0 <= ((NI) 32767))))) {
	goto LA6_;
}
if (nimMulInt(old_p0, ((NI) 2), (&TM__Q5wkpxktOdTGvlSRo9bzt9aw_13))) {
	raiseOverflow();
	goto BeforeRet_;
}
result = ((NI) TM__Q5wkpxktOdTGvlSRo9bzt9aw_13);
}
goto LA1_;
LA6_: ;
{
	NI TM__Q5wkpxktOdTGvlSRo9bzt9aw_14;
if (nimAddInt(((NI) ((old_p0) / (((NI) 2)))), old_p0, (&TM__Q5wkpxktOdTGvlSRo9bzt9aw_14))) {
	raiseOverflow();
	goto BeforeRet_;
}
result = ((NI) TM__Q5wkpxktOdTGvlSRo9bzt9aw_14);
}
LA1_: ;
}
	BeforeRet_: ;
		return result;
}
```
The `nimMulInt`/`nimAddInt`/`raiseOverflow` codepaths don't appear to actually quit the program, so do progress to ordinary function exit, but using `result` in an uninitialized way.


### Current Output

```text
$ gcc -c -Wmaybe-uninitialized -O1 @psystem.nim.c
@psystem.nim.c: In function ‘prepareSeqAddUninit’:
@psystem.nim.c:2199:10: warning: ‘result’ may be used uninitialized [-Wmaybe-uninitialize]
 2199 | newCap_1 = ((T9_ >= ((NI) TM__Q5wkpxktOdTGvlSRo9bzt9aw_15)) ? T9_ : ((NI) TM__Q5wkpxktOdTGvlSRo9bzt9aw_15));
      | ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@psystem.nim.c:2102:12: note: ‘result’ was declared here
 2102 |         NI result;
      |            ^~~~~~
@psystem.nim.c: In function ‘prepareAdd’:
@psystem.nim.c:3254:10: warning: ‘result’ may be used uninitialized [-Wmaybe-uninitialize]
 3254 | newCap_1 = ((newLen_1 >= T20_) ? newLen_1 : T20_);
      | ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@psystem.nim.c:2102:12: note: ‘result’ was declared here
 2102 |         NI result;
      |            ^~~~~~
@psystem.nim.c: In function ‘setLengthStrV2’:
@psystem.nim.c:3525:10: warning: ‘result’ may be used uninitialized [-Wmaybe-uninitialize]
 3525 | newCap_1 = ((newLen_p1 >= T30_) ? newLen_p1 : T30_);
      | ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@psystem.nim.c:2102:12: note: ‘result’ was declared here
 2102 |         NI result;
      |            ^~~~~~

$ clang -c -Wuninitialized -O1 @psystem.nim.c
@psystem.nim.c:2127:5: warning: variable 'result' is used uninitialized whenever 'if' condition is true [-Wsometimes-uninitialized]
 2127 | if (nimAddInt(((NI) ((old_p0) / (((NI) 2)))), old_p0, (&TM__Q5wkpxktOdTGvlSRo9bzt9aw_14))) {
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./nimbase.h:611:34: note: expanded from macro 'nimAddInt'
  611 |     #define nimAddInt(a, b, res) __builtin_saddll_overflow(a, b, (long long int*)res)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@psystem.nim.c:2136:10: note: uninitialized use occurs here
 2136 |                 return result;
      |                        ^~~~~~
@psystem.nim.c:2127:1: note: remove the 'if' if its condition is always false
 2127 | if (nimAddInt(((NI) ((old_p0) / (((NI) 2)))), old_p0, (&TM__Q5wkpxktOdTGvlSRo9bzt9aw_14))) {
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2128 |         raiseOverflow();
      |         ~~~~~~~~~~~~~~~~
 2129 |         goto BeforeRet_;
      |         ~~~~~~~~~~~~~~~~
 2130 | }
      | ~
@psystem.nim.c:2117:5: warning: variable 'result' is used uninitialized whenever 'if' condition is true [-Wsometimes-uninitialized]
 2117 | if (nimMulInt(old_p0, ((NI) 2), (&TM__Q5wkpxktOdTGvlSRo9bzt9aw_13))) {
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./nimbase.h:613:34: note: expanded from macro 'nimMulInt'
  613 |     #define nimMulInt(a, b, res) __builtin_smulll_overflow(a, b, (long long int*)res)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@psystem.nim.c:2136:10: note: uninitialized use occurs here
 2136 |                 return result;
      |                        ^~~~~~
@psystem.nim.c:2117:1: note: remove the 'if' if its condition is always false
 2117 | if (nimMulInt(old_p0, ((NI) 2), (&TM__Q5wkpxktOdTGvlSRo9bzt9aw_13))) {
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2118 |         raiseOverflow();
      |         ~~~~~~~~~~~~~~~~
 2119 |         goto BeforeRet_;
      |         ~~~~~~~~~~~~~~~~
 2120 | }
      | ~
@psystem.nim.c:2102:11: note: initialize the variable 'result' to silence this warning
 2102 |         NI result;
      |                  ^
      |                   = 0
2 warnings generated.
```

### Expected Output

```text
That there isn't any usage, even in the exception codepaths, of uninitialized data.
```

### Known Workarounds

_No response_

### Additional Information

_No response_

--- a/compiler/cgen.nim
+++ b/compiler/cgen.nim
@@ -1241,6 +1241,7 @@ proc allPathsAsgnResult(p: BProc; n: PNode): InitResultEnum =
     elif n[0].kind == nkSym and
         n[0].sym.magic in {mUnaryMinusI..mAbsI, mAddI..mPred} and
           optOverflowCheck in p.config.options:
+      # arithmetic operations may raise exceptions
       result = InitRequired
     else:
       for i in 0..<n.safeLen:

--- a/compiler/cgen.nim
+++ b/compiler/cgen.nim
@@ -1238,6 +1238,10 @@ proc allPathsAsgnResult(p: BProc; n: PNode): InitResultEnum =
         (n[0].kind == nkSym and sfNoReturn in n[0].sym.flags):
       # requires initializations when encountering unreachable code
       result = InitRequired
+    elif n[0].kind == nkSym and
+        n[0].sym.magic in {mUnaryMinusI..mAbsI, mAddI..mPred} and
+          optOverflowCheck in p.config.options:
+      result = InitRequired
     else:
       for i in 0..<n.safeLen:
         allPathsInBranch(n[i])

